[MODULAR] Veegee Station

Squird5 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Months ago, there were many complaints about Ice Box. I mentioned maybe trying to port or make a map. I mentioned /vg/station's Box and people seemed fairly receptive so I began work on it. 

Effectively a familiar but different station that comes from an alternate timeline. I tried to keep it as close as possible to /vg/'s Box as possible and intend to let it evolve over time into it's own creature.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
I thought a new map might be fun, and even though I'm a bit late to the party with Journey having beaten me to Space Box due to my stopping work on this project for a few months, I still think it would be a fun mix up over what people tend to know. A map that's familiar and unfamiliar at the same time.

The Asteroid also allows for secondary mining site for miners as well as a primary dig site for xenoarcheologists as well as a 'safe' Toxins testing site that doesn't scare the crew every time a bomb goes off.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Veegee Station and it's accompanying Research and Mining Asteroid
add: Veegee variant shuttles as well as a (WIP) Trader shuttle
code: New areas for Security, Research Outpost and Tradepost
code: Vox friendly air alarms!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
![boggs](https://user-images.githubusercontent.com/78889774/125567286-c6490e6d-f7b1-47a8-9261-5cc097579e81.png)
![asteroi](https://user-images.githubusercontent.com/78889774/125567307-aa0e2e15-4192-494a-b650-6121a75f819f.png)

FlamingLily commented:
Not sure how I feel about centralised AI core, for balance reasons, but otherwise this looks really nice

thestubborn commented:
one demand/request/whatever

call it pomfstation

Fikou commented on _maps/shuttles/research_veegee.dmm:
```diff
@@ -0,0 +1,128 @@
+/area/shuttle/research)
+"g" = (
+/obj/machinery/light{
+	dir = 8
```
doesnt use the directional light subtypes

Squird5 replied:
A lot of the things, lights included I believe, didn't have directionals when I started the project. The Research and Mining shuttles are literally just Mining_Box with a different computer at the helm. If that's a major issue, I can go back and start cleaning up all the directionals, though I'm not sure how long that will end up taking.

Gandalf2k15 commented:
So we can absolutely TM this and test it for now but you'll need to replace all instances of windows that don't use spawners and make all non directional fixtures directional

Cheshify commented:
3 boxstations

Squird5 commented:
🆑 
Changed all the windows to Spawners as requested
Changed as many objects as possible to directional variants
Added new areas
-/area/security/medical _Security Medbay_
-/area/security/prison/perma _Permanant Confinement_
-/area/ai_monitored/ai_sat/mommi _MoMMI Nest_

hugmeorelse commented:
@unit0016 @KathrinBailey

KathrinBailey commented:
Do we really need **three** versions of Boxstation?
One of 'em is going to have to go.

unit0016 commented:
Since I've been summoned to comment;

_Codewise;_
Lot of the code is copypasted structure-wise, comments inside lists, the _third box variant_. Also needlessly removes half the mining maps from basemap.dm. Also no note of credit to /vg/ save for the on-the-nose name.

_Mapping-Wise;_
## Nigh all of the map, unsurprisingly - was designed for /vg/'s features and not all of it was properly adapted.
## For instance, botany doesn't work on atmos and doesn't need atmos pipes.
![image](https://user-images.githubusercontent.com/50649185/125712249-354c8bba-470e-43b9-9d40-35d8c57cd578.png)
## Speaking of atmos pipes.. hook up freezers and heaters to the scrubbers line!
![image](https://user-images.githubusercontent.com/50649185/125712974-be3f2f8c-fe57-4260-8859-d200695ae66d.png)
## Sectech is deprecated - use the armadyne vendor instead.
![image](https://user-images.githubusercontent.com/50649185/125712615-15e3a050-0056-424e-bb00-4a3ab3c6c810.png)
## Now that I think about it, the entire asteroid level seems badly thought out. We already have a mining level in lavaland/rockplanet/icemoon/etc.

### Moving to onstation.

## Armory atmosdiff. Please use the debug mapping tools in-game to check for roundstart atmos diffs.
![image](https://user-images.githubusercontent.com/50649185/125713266-7b51335b-75bf-456a-9bbc-19176385aec7.png)

## Ignoring the lack of the skyrat staple funny rooms, a problem in of itself, I guarantee at least one prisoner will be using those air scrubbers to kill themselves. Also, the giant thing of glass purely exists to stop prisoners from spearing into escape and seems redundant to our roundstart no escape policy. Also, that isn't plastic!
![image](https://user-images.githubusercontent.com/50649185/125713319-88d19898-49a4-485b-9086-f0d36362d118.png)

## Security jetpacks. Ignoring balance questions as per policy, Spawns on top of the suit storage unit and isn't the security variant.
![image](https://user-images.githubusercontent.com/50649185/125713505-86a10c38-e6dd-44bc-8d15-f566e460fe59.png)

## Nuclear bomb in the vault instead of the self-destruct. There's also a few generic lootdrops that do nothing but eat up a bit of memory roundstart in the safe.
![image](https://user-images.githubusercontent.com/50649185/125713665-55a0bd39-37f4-479a-aa62-0b413715786d.png)

## This. Also, the whiteship is basically guaranteed to collide with the tradeship.
![image](https://user-images.githubusercontent.com/50649185/125713816-9d788d9b-826e-40cd-8634-6c3392a06bf0.png)

## We don't have the mechanics to support this.
![image](https://user-images.githubusercontent.com/50649185/125713789-84fe76ae-69f8-44a8-9bab-9366ab3ead90.png)

## The FREE OMNILATHE and CIRCUIT IMPRINTER a spacewalk away from arrivals. Also the drone dispenser which has been long deprecated in favor of the exosuit fab's maint drone option.
![image](https://user-images.githubusercontent.com/50649185/125714004-7fb33d3c-da30-4e3c-bf09-cba4acd11e1a.png)

## The absolute mess that is engineering. /TG/ is working on making the singulo uncontainable, TEG deprecated, etcetera.
![image](https://user-images.githubusercontent.com/50649185/125714212-e08c677e-b395-4ffb-9461-37a1f2169a04.png)

## Roundstart station-goal shield sats.
![image](https://user-images.githubusercontent.com/50649185/125714226-e37f7367-8854-4991-9c40-6cdd8f044c34.png)

## This supermatter setup. Just in general.
![image](https://user-images.githubusercontent.com/50649185/125714261-34c88f60-174b-4d69-8d02-6895301dadb6.png)

## Telescience is deprecated.
![image](https://user-images.githubusercontent.com/50649185/125714357-1b233f92-3191-4897-a669-e59f73be473e.png)

## This xenobio lab. Also just in general.
![image](https://user-images.githubusercontent.com/50649185/125714406-afd9ebb7-c596-4e10-b2ba-b313263c85d8.png)

## Science has two variants of the lathe, please use only the techfab.
![image](https://user-images.githubusercontent.com/50649185/125714484-148f08c2-4e54-4877-9c12-7322bbfda7cc.png)

## Robotics circuit imprinter isn't department-locked.
![image](https://user-images.githubusercontent.com/50649185/125714529-2e28d3a7-271d-4e28-a26b-2c9a5b8fc2df.png)

## Medbay's lack of operating tables. Also plumbing with no synths connected, and roundstart limbgrower. Morgue also doesn't have a smartfridge for harvested organs.
![image](https://user-images.githubusercontent.com/50649185/125714601-e35429d8-e9b1-43d2-9c49-cd159815d72a.png)

## Free maint cult robes.
![image](https://user-images.githubusercontent.com/50649185/125714665-7f0d11fb-d77a-4327-a831-52033e894711.png)

## Stools with no custom dir.
![image](https://user-images.githubusercontent.com/50649185/125714834-68a7212f-67c4-4230-8b8c-f6271a7ba4ff.png)

## Spare ID should come from the safe and not be a guaranteed drop anymore.
![image](https://user-images.githubusercontent.com/50649185/125714872-04d80bea-6733-4686-b15d-e649d24fb7f6.png)

## This conveyor belt.
![image](https://user-images.githubusercontent.com/50649185/125714934-7eaa92a3-ed67-48d2-bdaa-a966070a8bf9.png)

## Medbay genetics.
![image](https://user-images.githubusercontent.com/50649185/125715072-71ab8581-f52e-4da5-862c-31f05c72612f.png)

## This is either a race for the shotgun or a race for bar access every shift.
![image](https://user-images.githubusercontent.com/50649185/125715203-0e8a3ed0-6dbc-4c0e-9f5b-df6b4f59fdd2.png)

## Why is this a subtype of fore maintenance specifically? Probably should've put this in with the code specific stuff but didn't notice until I booted SDMM.
![image](https://user-images.githubusercontent.com/50649185/125715288-77c7051b-1e76-4d3e-ae3a-261f03099824.png)

## All assistants start here. While humorous, this is going to cause problems and is very 'gamey'.
![image](https://user-images.githubusercontent.com/50649185/125715463-2a15ae1c-b5f4-4d21-938e-65a5498d2da2.png)


And finally, misc mapping things;
## Please use template_noop for areas/turfs not part of the actual structure on map templates, not space.
![image](https://user-images.githubusercontent.com/50649185/125712733-0786fc27-04bc-4794-ab98-a963bacb4e56.png)

## Near all of the map is in dire need of decoration and expansion in general. It feels like a direct /vg/ port, which it, at current, is. 

## It is worth nothing that /vg/ and /tg/ probably have different access codes for airlocks. It is also worth noting that a good majority of this map, especially maintenance, needs some serious de-soulification and prettying up to look dirtier.

moreover i am of the opinion that carthage should be destroyed.

KathrinBailey commented:
Alright, I'm not going to close it yet but the foot needs to go down.
## We already have two versions of Boxtation.
Sorry, but I can't even justify replacing Icebox with this. Then we'll have Spacebox, and Spacebox2.
It looks like this has been ported without any real thought to how the TG codebase works, or instead, just a grand lack of familiarity with in-game mechanics. Even if we ignore the unused content (genetics in medical etc) and all the missing content, this is more or less of a step _backwards_ into an older version of box for no real reason.
Whilst I do appreciate the nostalgia value it _might_ have, it isn't enough to support the map.
You're going to have to ditch the VG part and do significant changes to make it unique enough to not be "Spacebox2", and even then, I don't think you'll be able to change enough to justify replacing Icebox with another Spacebox _but slightly different_:tm:.
**I'd recommend not spending any more time on this, in all honesty.**
With how much you'll need to change to make this unique enough, you'd be better off just making your own map from scratch instead.

Squird5 commented:
Understandable. Wasn't really ever intending to replace any of the Box maps so much as have an alternative. Again, this was mostly a learning experience for me and the play test did really help with realizing some things to pay more attention to. I'll just at least post the last working version I've got for the time being with a lot of fixes to it. Maybe it can be used as a rare map or something? If not, I'll probably just start thinking of how to design up Frontier Station.

🆑 
Fixes:
-More Assistant spawn locations
-Engineering Access
-Service Access
-Telesci Disposals
-Medbay colors
-Janicart Keys
-Paramed Lockers
-Medihud glasses
-White medkits
-Service Techfab moved/Service Hall created
-Science Prolathe removed
-NTR Office+Locker
-Bridge door cycler links
-Station side Ore Machine
-Chemistry access for Doctors
-Changed some Maint loot
-Robotics ExoFabs rearranged
-Vault Ore Silo added
-Paramed access to departments (Hopefully?)
-TEMP removed Jukeboxes
-Removed Meteor Shields
-Reloaded tech storage
-Cuffs on Bridge
-Vendor in Sec Checkpoint removed
-Nerfed Mechanic's room
-HOS Suit for HOS
-Fuck the Sec EVA hatch
-Monster shield for Research Outpost
-Removed Chapel Gun Firelock
-Some fixes to Atmos pipes
-Changed security jetpacks
-Double checked all doors access
-Rearrange Kitchen
-Nerf MoMMI Nest
-Reinforced Outer Hull
-Removed Limb Grower
-Added blood filter to Surgery
-Rearranged Morgue
-Added Xenobio console
-Minor prison changes

KathrinBailey commented:
I'll close it for now, since your work will still be safe on your branch.
